TESTING.md

```diff
@@ -0,0 +1,99 @@
+
+```bash
+#!/bin/bash
+
+mkdir test-project
+cd test-project
+git init
+echo "FROM plus3it/tardigrade-ci:0.8.0" > Dockerfile
+echo "SHELL := /bin/bash\n" > Makefile
+echo 'include $(shell test -f .tardigrade-ci || curl -sSL -o .tardigrade-ci "https://raw.githubusercontent.com/plus3it/tardigrade-ci/master/bootstrap/Makefile.bootstrap"; echo .tardigrade-ci)' >> Makefile
+echo '# tardigrade-ci' > .gitignore
+echo '.tardigrade-ci' >> .gitignore
+echo 'tardigrade-ci/' >> .gitignore
+```
+
+2. Run the following docker command in the test-project directory:
+
+```bash
+docker run --rm -v $PWD:/workdir --workdir=/workdir --entrypoint make python:3 help
+```
+
+3. This should result in the creation of both .tardigrade-ci and /tardigrade-ci
+
+4. Run the following command to remove the tardigrade-ci subdirectory and bootstrap Makefile:
+
+```bash
+docker run --rm -v $PWD:/workdir --workdir=/workdir --entrypoint make python:3 clean
+```
+
+5. Go to parent directory of test-project and remove the directory for initial setup:
+
+```bash
+rm -rf test-project/
+```
+
+### Via Docker (tardigrade-ci image)
+
+Running a tardigrade-ci container should automatically put the user in developer mode and create only the .tardigrade-ci file in the local environment.
+
+1. Environment setup should contain the tardigrade-ci project in root directory as well as thee test-project from previous steps.
+
+2. In the test-project directory, run the following command:
+
+```bash
+docker run --rm -v $PWD:/workdir --workdir=/workdir --entrypoint make plus3it/tardigrade-ci:latest help
+```
+
+3. This should result in only the .tardigrade-ci file being created.
+
+4. Clean up the environment by removing the tardigrade-ci project and test-project directory
+
+
+### Via Makefile
+
+This option uses `make` to invoke the targets. However, you may invoke any make target directly, e.g. `make ec/lint`. Be aware if you do so, the make target will attempt to install the tools it requires to your system.
+
+1. Run the following bash script or manually create a test-project directory with the following configurations:
+
+```bash
+#!/bin/bash
+
+mkdir test-project
+cd test-project
+git init
+echo "FROM plus3it/tardigrade-ci:0.8.0" > Dockerfile
+echo "SHELL := /bin/bash\n" > Makefile
+echo 'include $(shell test -f .tardigrade-ci || curl -sSL -o .tardigrade-ci "https://raw.githubusercontent.com/plus3it/tardigrade-ci/master/bootstrap/Makefile.bootstrap"; echo .tardigrade-ci)' >> Makefile
+echo '# tardigrade-ci' > .gitignore
+echo '.tardigrade-ci' >> .gitignore
+echo 'tardigrade-ci/' >> .gitignore
+```
+
+2. In the test-project direcotry, run the following command:
+
+```bash
+make help
+```
+
+3. This should result in the creation of both .tardigrade-ci and /tardigrade-ci. If the tardigrade-ci project was in the parent folder of the current test-project directory, then only the .tardigrade-ci should be created.
```

Comment:
```suggestion
3. This should result in the creation of both `.tardigrade-ci` and `tardigrade-ci/`.
```

> If the tardigrade-ci project was in the parent folder of the current test-project directory, then only the .tardigrade-ci should be created.

this part should be a separate, explicit test. one test without the `tardigrade-ci/` project in a parent/sibling directory, and one test with it present.